litgpt/model.py

```diff
@@ -298,6 +298,20 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
         return self.proj(x)
 
 
+class Phi3MLP(nn.Module):
```

Comment:
It should be possible to not need this class at all and instead reshape the weights for LLaMAMLP in the checkpoint conversion.

Reply:
100% agree. I was thinking the same thing. Similar to OLMo, I was hoping to get it working first and then simplify from there.

Reply:
New models by Apple have if-else statements for this case: https://huggingface.co/apple/OpenELM-270M-Instruct/blob/main/modeling_openelm.py#L405-L462

For simplicity we definitely shouldn't do the same.